app/src/main/java/com/sschuraytz/charcoaldrawing/DrawingView.java

```diff
@@ -76,4 +85,26 @@ public void onDraw(Canvas canvas) {
     public Paint getPaint() {
         return paint;
     }
+
+    private void printTexturedCircle(float pointX, float pointY)
+    {
+        rand = new Random();
```

Comment:
There is no reason to create Random each time (and you aren't using it).
Create it once.